project/scenes/Main.gd

```diff
@@ -59,7 +104,44 @@ func show_score():
 	$scoredisplay.get_node("scorelabel").text = "SCORE: " + str(score)
 	
 
+func generate_obstacle():
+	# if obstacles.is_empty() or last_obstacle.position.x < distance + randi_range(300, 500):
+		var obstacle_type = obstacle_types[randi() % obstacle_types.size()]
+		var obs = obstacle_type.instantiate()
+		var obs_height = obs.get_node("Sprite2D").texture.get_height()
+		var obs_scale = obs.get_node("Sprite2D").scale
+		var obs_x = screen_size.x + distance + 100
+		var obs_y = screen_size.y - ground_height - (obs_height * obs_scale.y / 2) + 5
+		# obs.position = Vector2i(obs_x, obs_y)
+		last_obstacle = obs
+		add_obs(obs, obs_x, obs_y)
+		last_obs_time = obs_timeout
+		
+		if difficulty > 0:
+			if (randi() % 2) == 0:
+				obs = eagle_scene.instantiate()
+				obs_x = screen_size.x + distance + 100
+				obs_y = eagle_heights[randi() % eagle_heights.size()]
+				add_obs(obs, obs_x, obs_y)
+		
+func add_obs(obs, x, y):
+	obs.position = Vector2i(x, y)
+	obs.body_entered.connect(hit_obs)
+	add_child(obs)
+	obstacles.append(obs)
+	
+func hit_obs(body):
+	if body.name == "Monkey":
+		game_over()
+		
 
+func game_over():
+	get_tree().paused = true
+	$GameOver.show()
+	
+	# might have to be changed
+	player_on_camera = false;
+	
 
 
 
```

Comment:
can you remove the extra lines and only leave one newline at the end of the file?